app/src/main/java/com/epfl/drawyourpath/database/FireDatabase.kt

```diff
@@ -380,54 +424,93 @@ class FireDatabase : Database() {
      * @param data the data snapshot containing the user List
      * @return a list containing the userIds of the friends
      */
-    private fun transformFriendsList(data: DataSnapshot?): List<String>{
-        if(data == null){
+    private fun transformFriendsList(data: DataSnapshot?): List<String> {
+        if (data == null) {
             return emptyList()
         }
         val friendListUserIds = ArrayList<String>()
 
-        for(friend in data.children){
+        for (friend in data.children) {
             friendListUserIds.add(friend.key as String)
         }
         return friendListUserIds
     }
 
+    /**
+     * Helper function to obtain the runs history from the database of the user
+     * @param data the data snapshot containing the history
+     * @return a list containing the history of the runs of the user
+     */
+    private fun transformRunsHistory(data: DataSnapshot?): List<Run> {
+        if (data == null) {
+            return emptyList()
+        }
+        val runsHistory = ArrayList<Run>()
+
+        for (run in data.children) {
+            val points = ArrayList<LatLng>()
+            for (point in run.child("path").child("points").children) {
+                val lat = point.child("latitude").getValue(Double::class.java)
+                val lon = point.child("longitude").getValue(Double::class.java)
+                if (lat != null && lon != null) {
+                    points.add(LatLng(lat, lon))
+                }
```

Comment:
As it is not normal behavior to have a null point, it might be worth to log the problem.

```kotlin
android.util.Log.w("DYP", "A run had an invalid coordinate")
```

or something similar.

Reply:
Good idea! I log the problem if the point is invalid in the following way:
`android.util.Log.w(FireDatabase::class.java.name, "A point of a run has invalid coordinates => ignoring the point")`
The tag is the class name to identify where the issue comes from easily.